nixos/modules/services/networking/i2pd.nix

```diff
@@ -1,695 +1,434 @@
 { config, lib, pkgs, ... }:
 
 with lib;
```

Comment:
Some people complain about this `with lib;` usage... See https://github.com/NixOS/nixpkgs/issues/208242 . Perhaps you can replace that with `inherit (lib) ...`?

Reply:
I think this change is justified when multiple attrsets are imported globally, but in this case `lib` is the only one.

[From the same discussion:](https://github.com/NixOS/nixpkgs/issues/208242#issuecomment-1381792638)

> ...
> with lib in modules is a bit harder. Having it span over the entire file is not ideal but I understand why people are doing it because lib is used a lot in modules. Keeping it for options or functions locally which use a lot of lib is totally fine to me especially if it is more than 2 or 3 occurrences.
> 
> Inheriting function from lib near the top of a file is more busy work for me and not improving much. Why do I need to keep track which functions from lib are used? Why do I need to scroll up and manually update the list? Why not just use with lib or write lib. before the function?
> Those lines are often not updated when the usage of a function is removed and it is also hard to review by eye.
> ...

Reply:
First, can you tell me precisely which functions you are really using from `lib` in this 434-line file (with a spurious blank line at the end)?

Second, are you sure that those `lib.mdDoc` you substituted by `mdDoc` are really equivalent to `mdDoc`, and not `lib.lib.mdDoc`?
Justify your answer ("ofborg didn't complain" or any other variation of "it worked in my machine" are not answers).

Third, the Nix language is not the C preprocessor language, and `with lib` is not equivalent to `#include  <cstdlib>`.
Nix does not merely copy-pastes code , it effectively parses the whole `lib`, even if lazily.

Reply:
@AndersonTorres

* (1) IMHO prior to this point, `lib` is already cached while evaluating other things in `nixpkgs`, so this shouldn't take longer than querying a hashmap.

* (2,3) Sorry, I really don't think that explicit reinheritance of `lib`'s prelude can resolve anything, apart from being a hassle (in the context of a not so complicated module, and considering that `lib`'s prelude is the only global import there). There are 33 functions and 1 attrset accessed using `with lib;` to be precise.

* (4) `lib.mdDoc` was inserted tree-wide with little awareness of specifics of the module while migrating from plaintext. But even otherwise, attributes in the `lib`'s prelude are usually have distinct names (and signatures), so I don't think a real collision could take place unnoticed.

Btw, what is wrong with the blank line at the end of the file?